src/pybind/mgr/cephadm/module.py

```diff
@@ -2608,6 +2610,9 @@ def remove_service(self, service_name: str, force: bool = False) -> str:
                 raise OrchestratorError(
                     f'If {service_name} is removed then the following OSDs will remain, --force to proceed anyway\n{msg}')
 
+        if service_name == 'mgmt-gateway':
+            self.set_module_option('secure_monitoring_stack', False)
```

Comment:
We need to be really careful about documenting this sort of behavior. I don't think there are many, if any,other places where we start modifying user facing config options on our own. There's also nothing stopping the user from toggling this setting themselves whenever they want (e.g. they could turn it off after deploying the mgmt-gateway service).

Reply:
@adk3798 I changed the way we were checking for security to see if it's enabled or not. I introduced a new function in the mgr `_get_security_config` which returns the current status. Security is considered as enabled both when the `secure_monitoring_stack` is true or when the mgmt-gateway is deployed. This way, the user can't mess with the mgmt-gateway by changing the value of the `security_monitoring_stack` (at most changing its value would lead to a reconfig which will end up with the same config as previously). 

This has caused a lot of changes in the code but I think the new code handles better security config